src/app/accounts/login.component.ts

```diff
@@ -58,8 +61,10 @@ export class LoginComponent extends BaseLoginComponent implements OnDestroy {
         // @override by Cozy
         // check if code is run into a Cozy app
         // if yes, retrieve url and user email from the htlm
-        const cozyDataNode = document.getElementById('cozy-app');
-        const cozyDomain = cozyDataNode ? cozyDataNode.dataset.cozyDomain : null;
+        const cozyDataNode = document.querySelector('[role=application]') as HTMLElement;
```

Comment:
Do you already have a cozyClient instance here? If yes, you can remove this querySelector and the next line and do something like: 

const { cozyDomain } = this.clientService.GetClient().getInstanceOptions() . Like that you're not subject to the way we store the data within the HTML. It's more futur proof